cpp/dolfinx/fem/assemble_matrix_impl.cpp

```diff
@@ -56,56 +56,56 @@ void fem::impl::assemble_matrix(
   using type = fem::FormIntegrals::Type;
   for (int i = 0; i < integrals.num_integrals(type::cell); ++i)
   {
-    const auto & fn = integrals.get_tabulate_tensor(type::cell, i);
+    const auto& fn = integrals.get_tabulate_tensor(type::cell, i);
     const std::vector<std::int32_t>& active_cells
         = integrals.integral_domains(type::cell, i);
 
-    fem::impl::assemble_cells<IndexType, ScalarType>(
+    fem::impl::assemble_cells<ScalarType>(
         mat_set_values_local, mesh, active_cells, dofs0, num_dofs_per_cell0,
         dofs1, num_dofs_per_cell1, bc0, bc1, fn, coeffs, constant_values);
   }
 
   for (int i = 0; i < integrals.num_integrals(type::exterior_facet); ++i)
   {
-    const auto & fn = integrals.get_tabulate_tensor(type::exterior_facet, i);
+    const auto& fn = integrals.get_tabulate_tensor(type::exterior_facet, i);
     const std::vector<std::int32_t>& active_facets
         = integrals.integral_domains(type::exterior_facet, i);
-    fem::impl::assemble_exterior_facets<IndexType, ScalarType>(
+    fem::impl::assemble_exterior_facets<ScalarType>(
         mat_set_values_local, mesh, active_facets, dofmap0, dofmap1, bc0, bc1,
         fn, coeffs, constant_values);
   }
 
   for (int i = 0; i < integrals.num_integrals(type::interior_facet); ++i)
   {
     const std::vector<int> c_offsets = a.coefficients().offsets();
-    const auto & fn = integrals.get_tabulate_tensor(type::interior_facet, i);
+    const auto& fn = integrals.get_tabulate_tensor(type::interior_facet, i);
     const std::vector<std::int32_t>& active_facets
         = integrals.integral_domains(type::interior_facet, i);
-    fem::impl::assemble_interior_facets<IndexType, ScalarType>(
+    fem::impl::assemble_interior_facets<ScalarType>(
         mat_set_values_local, mesh, active_facets, dofmap0, dofmap1, bc0, bc1,
         fn, coeffs, c_offsets, constant_values);
   }
 }
 //-----------------------------------------------------------------------------
 // \cond doxygen should ignore
```

Comment:
I think we can remove these doxygen statements now that the test image has a more recent doxygen.